### PR TITLE
refactor: squad ui improvements

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- There was a lot of implementation that was inaccurate with the design spec. So there were a lot of spacing changes.
- I have also refactored some areas where there were unnecessary nested containers.
- The intended changes from the design team are:
  - The "Create new post" button is now fully white (primary)
  - Invite button:
    - We add "Invite" wording next and changed the icon to +
    - Reduce the avatars preview on mobile to only 3
  - On mobile view:
    - We centered the entire top information
    - We removed the icon from the feedback button

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1078 #done

### CODE REVIEW

File: packages/shared/src/components/squads/SquadHeaderBar.tsx [hunk 1]
```diff
@@ -45,48 +45,50 @@ export function SquadHeaderBar({
...
+        'flex flex-row flex-wrap items-center gap-4',
+        className,
+      )}
+    >
+      <div className="flex overflow-hidden items-center rounded-14 border border-theme-divider-secondary">
         <SimpleTooltip placement="top" content="Members list">
-          <Button
-            className="active:bg-theme-active btn-tertiary !pl-1 !pr-3 !rounded-r-none"
-            buttonSize="medium"
+          <button
```
Comment (idoshamun) [inline packages/shared/src/components/squads/SquadHeaderBar.tsx:57]: Why the change?

Comment (sshanzel) [reply packages/shared/src/components/squads/SquadHeaderBar.tsx:57]: There were a lot of overrides on the base styles of the button itself which lost the purpose of using the component itself.

**Review by idoshamun — APPROVED** [review]
Perfect!
